Sources/Clickstream/Dependency/Clickstream/Event/ClickstreamEvent.swift

```diff
@@ -136,7 +136,7 @@ class ClickstreamEvent: AnalyticsPropertiesModel {
     }
 
     static func == (lhs: ClickstreamEvent, rhs: ClickstreamEvent) -> Bool {
-        lhs.toJson() == rhs.toJson()
+        lhs.toJsonObject().toJsonString() == rhs.toJsonObject().toJsonString()
```

Comment:
While this approach might seem convenient, it can lead to unexpected behavior due to the following reasons:

- Loss of data: The JSON representation might not capture all the data contained in the ClickstreamEvent objects. For example, custom properties or internal fields might not be serialized to JSON. This could lead to false positives or negatives when comparing events.

- Ordering sensitivity: The order of elements within the JSON object might not be preserved when converting it to a string. This could lead to different JSON strings being generated for the same ClickstreamEvent object, depending on the implementation of the toJsonObject method. This inconsistency can make it difficult to reliably compare events.

- Floating-point precision: If the ClickstreamEvent objects contain floating-point numbers, converting them to JSON and back might introduce rounding errors. This could lead to events being considered unequal even if their values are essentially the same.

Therefore, it is recommended to compare ClickstreamEvent objects directly by comparing their individual properties, such as event_type, timestamp, user_id, and item_id. This approach is more reliable and avoids the pitfalls of using JSON representations.

Reply:
Yes, this is indeed a way to simplify the code for calculate two ClickstreamEvent object whether equals. Moreover, this is not fast enough to execute and will generate redundant string objects. It is indeed better to compare each attribute directly here.
After reviewing the code, this method is also redundant in new version and it should be removed.

Regarding the three issues mentioned:
* Loss of data: This method is indeed not able to compare all potential variables. Here, all attributes that need to be uploaded are compared. It should be considered that the two events are consistent in terms of business logic.
* Ordering sensitivity:  the method [`toJsonObject` ](https://github.com/awslabs/clickstream-swift/blob/5b921d2483bf3da110637e61c5b118c5813ffd99/Sources/Clickstream/Dependency/Clickstream/Event/ClickstreamEvent.swift#L82)is a customize method for ClickstreamEvent and the order is specified explicitly in the extension [`toJsonString()`](https://github.com/awslabs/clickstream-swift/blob/5b921d2483bf3da110637e61c5b118c5813ffd99/Sources/Clickstream/Support/Extension/JsonObject%2BToJsonString.swift#L13) method of JsonObject, so the order here is stable.
* Floating-point precision: before we convert the object to json we have already handle the [precision problem in double type](https://github.com/awslabs/clickstream-swift/blob/5b921d2483bf3da110637e61c5b118c5813ffd99/Sources/Clickstream/Dependency/Clickstream/Event/ClickstreamEvent.swift#L129-L133), and ensure that the value is equals to origin value and every conversion is consistent.